libmscore/textedit.cpp

```diff
@@ -122,6 +122,13 @@ void TextBase::endEdit(EditData& ed)
                   newlyAdded = true;
                   undo->mergeCommands(ted->startUndoIdx - 1);
                   }
+            else {
+                  IF_ASSERT_FAILED(parent()) {
+                      return;
+                  }
+                  
+                  newlyAdded = (Ms::toTBox(parent()) != nullptr);
```

Comment:
Actually my suggestion was more about applying also the `undo->mergeCommands()` part above to this case, this seems to be a more correct thing to do if `TBox` text was edited. Also it is not completely correct to consider the text newly added if old text is empty: with text frame you can clear the content of this text, leave edit mode and start editing it after that. It would be more correct to determine if the text was added as a part of a `TBox` by something like `parent()->isTBox() && ucmd->hasFilteredChildren(Filter::AddElement, parent())`, this could probably just be added to the `if` condition above.

However when writing that comment in the issue I didn't also really consider the case when the text in the end of editing process is empty: with `TBox`, given its possibility to have an empty text, it is probably even better not to remove this text at all. Therefore the condition `if (actualPlainText.isEmpty())` below should probably also be modified to something like `if (actualPlainText.isEmpty() && !parent()->isTBox())`.

I can create a pull request with such changes but they would also need some testing.

Reply:
Created a pull request here: #6141